File: apps/monolith/src/main/java/com/example/iter/admin/stats/dto/response/AdminStatsResponse.java
package com.example.iter.admin.stats.dto.response;

public record AdminStatsResponse(
        long userCount,
        long equipmentCount,
        long unresolvedReportCount,
        long paymentCount
) {
    public static AdminStatsResponse of(
            long userCount,
            long equipmentCount,
            long unresolvedReportCount,
            long paymentCount
    ) {
        return new AdminStatsResponse(userCount, equipmentCount, unresolvedReportCount, paymentCount);
    }
}

File: apps/monolith/src/main/kotlin/com/example/iter/admin/stats/dto/response/AdminStatsResponse.kt
package com.example.iter.admin.stats.dto.response

import kotlin.jvm.JvmRecord

@JvmRecord
data class AdminStatsResponse(val userCount: Long, val equipmentCount: Long, val unresolvedReportCount: Long, val paymentCount: Long) {
    companion object {
        @JvmStatic
        fun of(userCount: Long, equipmentCount: Long, unresolvedReportCount: Long, paymentCount: Long): AdminStatsResponse = AdminStatsResponse(
            userCount,
            equipmentCount,
            unresolvedReportCount,
            paymentCount
        )
    }
}
